Add 'mailservice' LDAP schema

drybjed opened the pull request:
Fixes #1128

drybjed committed:
[slapd] Add the 'mailservice' LDAP schema

drybjed committed:
[docs] Add missing reference to 'nextcloud' schema

reixd commented:
+1 approved from my side.
Thank you, this schema is more generic than my proposal.

drybjed committed:
[slapd] Extend the 'mailservice' LDAP schema

drybjed committed:
[slapd] Ensure that e-mail addresses are unique

The 'mail' LDAP attribute will be used as an uniqueness constraint
across LDAP entriess for the 'mailAddress' and 'mailAlternateAddress'
attributes.

drybjed committed:
[slapd] Index new e-mail LDAP attributes
